[clang-format] Add 'cl' to enable OpenCL kernel file formatting

wenju-he opened the pull request:
There are many .cl files in llvm repo. It would be great that clang-format can support it.

wenju-he committed:
[clang-format] Add 'cl' to enable OpenCL kernel file formatting

mydeveloperday reviewed (commented):
I feel like there are more places where this needs to be addressed if we are going to add it.

mydeveloperday commented:
i.e. clang-format-diff.py

mydeveloperday commented:
should we consider .ocl?

wenju-he committed:
return C language format style for OpenCL

wenju-he commented:
> i.e. clang-format-diff.py

Thanks. I just find out that OpenCL support was added in clang-format-diff.py by https://github.com/llvm/llvm-project/commit/c49406d18c84b5b0c4f90703e162d9496cf28db6

wenju-he commented:
> I feel like there are more places where this needs to be addressed if we are going to add it.

I added support in language detection for OpenCL in https://github.com/llvm/llvm-project/pull/134529/commits/88c11747fcc8db1921dfd8f73c9330c662f7fd91. It returns C language since OpenCL is based on C99 and C11 with restrictions, see https://registry.khronos.org/OpenCL/specs/3.0-unified/html/OpenCL_C.html

wenju-he commented:
> should we consider .ocl?

.ocl isn't used as file extension for OpenCL. See https://github.com/llvm/llvm-project/blob/a38ad6e2a24b46eaea481c04de566dd2a45d667b/clang/lib/Frontend/FrontendOptions.cpp#L31

owenca commented:
> > I feel like there are more places where this needs to be addressed if we are going to add it.
> 
> I added support in language detection for OpenCL in [88c1174](https://github.com/llvm/llvm-project/commit/88c11747fcc8db1921dfd8f73c9330c662f7fd91). It returns C language since OpenCL is based on C99 and C11 with restrictions, see https://registry.khronos.org/OpenCL/specs/3.0-unified/html/OpenCL_C.html

OpenCL is formatted as C++ now. Do we really need to change that? See #132832 for potential problems.

wenju-he committed:
Revert "return C language format style for OpenCL"

This reverts commit 88c11747fcc8db1921dfd8f73c9330c662f7fd91.

wenju-he commented:
> > > I feel like there are more places where this needs to be addressed if we are going to add it.
> > 
> > 
> > I added support in language detection for OpenCL in [88c1174](https://github.com/llvm/llvm-project/commit/88c11747fcc8db1921dfd8f73c9330c662f7fd91). It returns C language since OpenCL is based on C99 and C11 with restrictions, see https://registry.khronos.org/OpenCL/specs/3.0-unified/html/OpenCL_C.html
> 
> OpenCL is formatted as C++ now. Do we really need to change that? See #132832 for potential problems.

reverted. Thanks. Default should be better if it works better.

owenca commented on clang/tools/clang-format/git-clang-format:
```diff
@@ -126,6 +126,7 @@ def main():
             "pb.txt",
             "textproto",
             "asciipb",  # TextProto
+            "cl", # OpenCL
```
Move this line to just below line 108 above? Do we want to add `"clcpp", # OpenCL C++`? If yes, we should add it to clang-format-diff.py too.

wenju-he replied:
done, moved after line 108. 
> Do we want to add "clcpp", # OpenCL C++?

I actually don't know about the current development status of C++ for OpenCL in clang. So If there is such request, I think it can be added in a separate PR.

wenju-he committed:
move after line 108

wenju-he commented:
@owenca could you please help me to merge? Thanks. I don't have commit permission.
